### DISCUSSION TITLE
Slow writting when Grammarly is enabled

### DISCUSSION THREAD

**Discussion by cnicodeme** [ueberdosis/tiptap#3706] [General Questions] [answered]
Hi everyone!

When I write some text in a Tiptap editor, after 2 paragraph, the text starts to appear more slowly than what I write (by a bunch of 5/10 letters at a time, and it gets worse the more I write).

I initially thought it was an issue with my computer being slow or my implementation of Tiptap being to resource-hungry, but as soon as I disabled Grammarly, everything is fine, and I have no issues at all.

Am I the only one having this or does anyone here had a similar issue?

If you have, did you find a way to have Grammarly not slow down the editor ?

Thanks for your help!

**Comment by francistogram** [comment]
Did you end up figuring out a solution to this @cnicodeme?

**Reply by cnicodeme** [reply]
Not really, I stopped using Grammarly 😅

**Reply by francistogram** [reply]
hahaha fair fair. We're running into this issue with our users who use Grammarly 😆 

Thanks anyways!

**Reply by jeffparaform** [reply]
ah we're running into this issue now too. did you find a workaround?

**Reply by francistogram** [reply, marked as answer]
Not a great solution but been adding

```
data-gramm="false"
data-gramm_editor="false"
data-enable-grammarly="false"
```

to our text fields. That being said I've noticed that Grammarly works fine on TipTap's example editor so I suspect I'm doing something wrong and it's triggering a billion re-renders or something like that

**Reply by bdbch** [reply]
Most performance issues come from to many rerenders being fired. Tiptap 3.0 should improve this a little bit as right now Tiptap rerenders everytime a transaction is fired.

Disabling grammarly would work as @francistogram proposed I guess, never used it myself before.

**Reply by francistogram** [reply]
A few followups here, hopefully helps someone
1. I realized that I didn't set `lang="en"` anywhere in my app and surprisingly that's what was causing Grammarly to freeze (I assume trying to figure out the language). This was the real culprit to my issue
2. I couldn't get Grammarly to disable on TipTap with those fields and Grammarly support said to try `grammarly-desktop-disable=true`
3. Worst worst case if nothing fixes it for you, you can go through Grammarly support support@grammarly.zendesk.com and ask them to disable Grammarly for your whole domain
